purchase_stock_picking_return_invoicing/models/account_invoice.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services
+#           <contact@eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+from openerp import models
+_logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    def _prepare_invoice_line_from_po_line(self, line):
+        data = super(AccountInvoice,
+                     self)._prepare_invoice_line_from_po_line(line)
```

Comment:
Why not you use
if data:
    if line.product_id.....

by this way you avoid the OCB Error